### PR TITLE
Fixed #28411 -- Used cx_Oracle.Cursor.outputtypehandler instead of _rowfactory() on Oracle.

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/28411

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: django/db/backends/oracle/base.py [hunk 1]
```diff
@@ -396,17 +396,46 @@ def __init__(self, connection):
         self.cursor.arraysize = 100
 
     @staticmethod
+    def _output_number_converter(value):
+        if '.' in value:
```
Comment (timgraham) [inline django/db/backends/oracle/base.py]: I'd use `return a if b else c` but it's up to you.

File: django/db/backends/oracle/base.py [hunk 2]
```diff
@@ -396,17 +396,46 @@ def __init__(self, connection):
...
+                    # This will normally be an integer from a sequence,
+                    # but it could be a decimal value.
+                    outconverter = FormatStylePlaceholderCursor._output_number_converter
+                else:
+                    # FLOAT column: binary-precision floating point.
+                    # This comes from FloatField columns.
+                    outconverter = float
+            elif precision > 0:
+                # NUMBER(p,s) column: decimal-precision fixed point.
+                # This comes from IntField and DecimalField columns.
```
Comment (timgraham) [inline django/db/backends/oracle/base.py]: update to "IntegerField"?

File: django/db/backends/oracle/base.py [hunk 3]
```diff
@@ -396,17 +396,46 @@ def __init__(self, connection):
...
+    @staticmethod
     def _output_type_handler(cursor, name, defaultType, length, precision, scale):
         """
         Called for each db column fetched from cursors. Return numbers as
         strings so that decimal values don't have rounding error.
         """
         if defaultType == Database.NUMBER:
+            if scale == -127:
+                if precision == 0:
+                    # NUMBER column: decimal-precision floating point
```
Comment (timgraham) [inline django/db/backends/oracle/base.py]: add period

File: django/db/backends/oracle/base.py [hunk 4]
```diff
@@ -396,17 +396,46 @@ def __init__(self, connection):
...
+                    # but it could be a decimal value.
+                    outconverter = FormatStylePlaceholderCursor._output_number_converter
+                else:
+                    # FLOAT column: binary-precision floating point.
+                    # This comes from FloatField columns.
+                    outconverter = float
+            elif precision > 0:
+                # NUMBER(p,s) column: decimal-precision fixed point.
+                # This comes from IntField and DecimalField columns.
+                if scale == 0:
```
Comment (timgraham) [inline django/db/backends/oracle/base.py]: outconverter = a if b else c?

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**